libnavui-androidauto/src/androidTest/java/com/mapbox/androidauto/testing/BitmapTestUtil.kt

```diff
@@ -71,26 +76,21 @@ class BitmapTestUtil(
     fun assertBitmapsSimilar(testName: TestName, actual: Bitmap) {
         val filename = testName.methodName + ".png"
         val expectedBitmapFile = expectedAssetsDirectoryName + File.separator + filename
-        try {
-            context.assets.open(expectedBitmapFile).use {
-                val expected = BitmapFactory.decodeStream(it)
-                val difference = calculateDifference(expected, actual)
-                // If the images are different, write them to a file so they can be uploaded for
-                // debugging.
-                if (difference.similarity > 0.01) {
-                    writeBitmapFile(testName, actual)
-                    writeBitmapFile(
-                        "${testName.methodName}-diff", difference.difference
-                    )
-                    fail(
-                        "The ${testName.methodName} image failed with similarity: " +
-                            "${difference.similarity}"
-                    )
-                }
+        context.assets.open(expectedBitmapFile).use {
+            val expected = BitmapFactory.decodeStream(it)
+            val difference = calculateDifference(expected, actual)
+            // If the images are different, write them to a file so they can be uploaded for
+            // debugging.
+            if (difference.similarity > 0.01) {
+                writeBitmapFile(testName, actual)
+                writeBitmapFile(
+                    "${testName.methodName}-diff", difference.difference
+                )
+                fail(
+                    "The ${testName.methodName} image failed with similarity: " +
+                        "${difference.similarity}"
+                )
             }
-        } catch (t: Throwable) {
-            writeBitmapFile(testName, actual)
```

Comment:
This try catch was not an accident https://github.com/mapbox/1tap-android/pull/1274#discussion_r732896640

There's a better way to do this (writing to a file inside a catch is in general a bad idea).. But it's catching the case when "expected" bitmap is missing

Reply:
Either way it will throw an exception, so I think we should add it back in. 

It makes it a little easier to write new tests because your files are uploaded

Reply:
`open(@NonNull String fileName) throws IOException`

It's the `assets.open`. An improvement could be to change it to `} catch (t: IOException) {`

Reply:
Oh, I see. I removed it because the fail method could provoke an exception and then we'd have 2 "actual" files.
I placed only the reading of expected into try-catch.